### PR TITLE
Homogenize IEEE 802.15.4 channel configuration

### PR DESCRIPTION
**Pull request by simonduq**
Introduces `IEEE802154_CONF_DEFAULT_CHANNEL` and `IEEE802154_DEFAULT_CHANNEL` for easier configuration of the default channel across platforms. Also enables adding to `viewconf` and to initial printouts from `main`.

### CODE REVIEW

File: arch/cpu/cc26xx-cc13xx/cc13xx-cc26xx-conf.h [hunk 1]
```diff
@@ -81,10 +73,6 @@
 #if CC13XX_CONF_PROP_MODE
 #define NETSTACK_CONF_RADIO        prop_mode_driver
 
-#ifndef RF_CORE_CONF_CHANNEL
-#define RF_CORE_CONF_CHANNEL                      0
```
Comment (simonduq) [inline arch/cpu/cc26xx-cc13xx/cc13xx-cc26xx-conf.h]: I'm not sure about this one. Did we need the channel set to 0 here?

Comment (g-oikonomou) [reply arch/cpu/cc26xx-cc13xx/cc13xx-cc26xx-conf.h]: Yes please, channels count from 0 upwards in .15.4g

File: examples/coap/project-conf.h [hunk 2]
```diff
@@ -40,7 +40,7 @@
 #define PROJECT_ERBIUM_CONF_H_
 
 /* Custom channel and PAN ID configuration for your project. */
-/* #define RF_CHANNEL                    26 */
+/* #define IEEE802154_CONF_DEFAULT_CHANNEL                    26 */
 /* #define IEEE802154_CONF_PANID     0xABCD */
```
Comment (g-oikonomou) [inline examples/coap/project-conf.h]: I wonder if we should we get rid of those

File: os/contiki-main.c [hunk 3]
```diff
@@ -86,9 +86,17 @@ main(void)
   platform_init_stage_two();
 
   LOG_INFO("Starting " CONTIKI_VERSION_STRING "\n");
-
-  LOG_INFO(" Net: %s\n", NETSTACK_NETWORK.name);
-  LOG_INFO(" MAC: %s\n", NETSTACK_MAC.name);
+#define TSCH_HOPPING_SEQUENCE_16_16 (uint8_t[]){ 16, 17, 23, 18, 26, 15, 25, 22, 19, 11, 12, 13, 24, 14, 20, 21 }
```
Comment (g-oikonomou) [inline os/contiki-main.c]: I am not hugely keen on defines intermingled with code.

File: os/contiki-main.c [hunk 4]
```diff
@@ -86,9 +86,17 @@ main(void)
...
+  LOG_INFO("- Routing: %s\n", NETSTACK_ROUTING.name);
+  LOG_INFO("- Net: %s\n", NETSTACK_NETWORK.name);
+  LOG_INFO("- MAC: %s\n", NETSTACK_MAC.name);
+  LOG_INFO("- 802.15.4 PANID: 0x%04x\n", IEEE802154_PANID);
+#if MAC_CONF_WITH_CSMA
+  LOG_INFO("- 802.15.4 Channel: %u\n", IEEE802154_DEFAULT_CHANNEL);
+#endif /* MAC_CONF_WITH_CSMA */
+#if MAC_CONF_WITH_TSCH
```
Comment (g-oikonomou) [inline os/contiki-main.c]: Surely best use `#elif` here?

File: os/net/mac/mac.h [hunk 5]
```diff
@@ -43,6 +43,14 @@
...
 #include "dev/radio.h"
 
+/**
+ *\brief The default channel for IEEE 802.15.4 networks.
+ */
+#ifdef IEEE802154_CONF_DEFAULT_CHANNEL
+#define IEEE802154_DEFAULT_CHANNEL           IEEE802154_CONF_DEFAULT_CHANNEL
+#else /* IEEE802154_CONF_DEFAULT_CHANNEL */
+#define IEEE802154_DEFAULT_CHANNEL           26
+#endif /* IEEE802154_CONF_DEFAULT_CHANNEL */
```
Comment (g-oikonomou) [inline os/net/mac/mac.h]: It's not immediately clear to me why we need this here as well as in `os/contiki-default-conf.h`?

File: os/contiki-main.c (hunk 3 above)
Comment (simonduq) [reply os/contiki-main.c]: oops this was leftover from a test, never intended for commit!

File: os/contiki-main.c (hunk 4 above)
Comment (simonduq) [reply os/contiki-main.c]: sure

**Comment by g-oikonomou** [comment]
Good stuff here!

File: os/net/mac/mac.h (hunk 5 above)
Comment (simonduq) [reply os/net/mac/mac.h]: Are you suggesting keeping only the `_CONF` version of the flag everywhere, or rather having `contiki-default-conf.h` define the non-`_CONF` version? (currently if only defines `_CONF` flags)

File: examples/coap/project-conf.h (hunk 2 above)
Comment (simonduq) [reply examples/coap/project-conf.h]: agree, will remove

File: arch/cpu/cc26xx-cc13xx/cc13xx-cc26xx-conf.h (hunk 1 above)
Comment (simonduq) [reply arch/cpu/cc26xx-cc13xx/cc13xx-cc26xx-conf.h]: ok

**Comment by simonduq** [comment]
Thanks for the review!

File: os/net/mac/mac.h (hunk 5 above)
Comment (g-oikonomou) [reply os/net/mac/mac.h]: Never mind, all good

**Comment by simonduq** [comment]
Comments should be addressed now

**Pull request merged**